src/code_8048480.c

```diff
@@ -294,3 +348,433 @@ void sub_80487CC(struct Entity *pokemon, struct Entity * target, u32 param_3, u3
 {
     sub_8078B5C(pokemon, target, param_3, param_4, 1);
 }
+
+static inline bool8 sub_80487E0_sub(struct EntityInfo *pokemonInfo)
+{
+    if (pokemonInfo->joinedAt == 0x4A)
```

Comment:
Dungeon IDs are defined in `constants/dungeon.h`. 0x47 is `DUNGEON_RESCUE_TEAM_BASE`, while 0x4A is `DUNGEON_JOIN_LOCATION_CLIENT_POKEMON`.